### PR TITLE
Add lodf functions

### PR DESCRIPTION
**Pull request by alefcastelli**
Created all the missing LODF and VirtualLODF function.

Took the chance to fix some bugs and add functions to PTDF and VirtualPTDF for the sake of coherence.

Fixed docstring in different files.

Changed getindex for ABA matrix such that if a reference bus is selected an error is thrown

### CODE REVIEW

File: src/lodf_calculations.jl
Comment (jd-lara) [inline src/lodf_calculations.jl:103]:
```suggestion
    lodf_t[SparseArrays.diagind(lodf_t)] .= -1.0
```

File: src/lodf_calculations.jl
Comment (jd-lara) [inline src/lodf_calculations.jl:126]:
```suggestion
    lodf_t[SparseArrays.diagind(lodf_t)] .= -1.0
```

File: src/row_cache.jl
Comment (jd-lara) [inline src/row_cache.jl:120]: having a Union here in the type annotation doesn't help. Can be eliminated.

File: src/row_cache.jl
Comment (jd-lara) [inline src/row_cache.jl:102]:
```suggestion
function Base.setindex!(
    cache::RowCache{T},
    val::T,
    key::Int,
) where  T <: Union{Vector{Float64}, SparseArrays.SparseVector{Float64}}
```

File: src/row_cache.jl
Comment (jd-lara) [inline src/row_cache.jl:120]:
```suggestion
    cache::RowCache{T},
    key::Int,
)::T
```

File: src/virtual_lodf_calculations.jl
Comment (jd-lara) [inline src/virtual_lodf_calculations.jl:67]: I don't know if we should have this function. I think it might be an awkward workflow since it assumes that the user wants to sparsity after having created the rows. It's better we don't support this.

File: src/virtual_ptdf_calculations.jl
Comment (jd-lara) [inline src/virtual_ptdf_calculations.jl:156]: Same here

**Review by jd-lara — CHANGES_REQUESTED** [review]
There are some minor things to change before this PR can be merged. Please refer to the comments.

File: src/virtual_lodf_calculations.jl
Comment (alefcastelli) [reply src/virtual_lodf_calculations.jl:67]: ok, I will remove this function

File: src/virtual_ptdf_calculations.jl
Comment (alefcastelli) [reply src/virtual_ptdf_calculations.jl:156]: it'll be removed

**Comment by alefcastelli** [comment]
with the latest commit everything should be solved. Waiting for review and then the branch can be merged.

**Pull request merged**
